setup-node-env/action.yml

```
@@ -4,21 +4,39 @@
 
 name: Setup Node
 
-description: Setups up Node.js, restores NPM cache and installs dependencies
+author: Gabriel Montes
+
+description: Setups up Node, restores NPM cache and installs dependencies
 
 inputs:
+  cache:
+    description: "Used to specify a package manager for caching in the default directory. Supported values: npm, yarn, pnpm."
+    default: npm
   node-version:
-    description: The version of Node to install.
+    description: The version of Node to install
     default: ""
 
 runs:
   using: composite
   steps:
-    - uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
+    - uses: pnpm/action-setup@v4.0.0
```

As this is a third-party action, not maintained by either GitHub or Docker, we should pin it to a commit hash in order to make it immutable.

Also, a minor update is available: https://github.com/pnpm/action-setup/releases/tag/v4.1.0 (changes: https://github.com/pnpm/action-setup/compare/v4.0.0...v4.1.0)

```suggestion
      - uses: pnpm/action-setup@a7487c7e89a18df4991f7f222e4898a00d66ddda # v4.1.0
```

https://github.com/pnpm/action-setup/releases/tag/v4.1.0
https://github.com/pnpm/action-setup/commit/a7487c7e89a18df4991f7f222e4898a00d66ddda